python/test/snowflake/test_snowflake_to_gcs.py

```diff
@@ -0,0 +1,328 @@
+        mock_spark_session.dataframe.DataFrame.write.mode().partitionBy().option.assert_called_with(constants.HEADER, True)
+        mock_spark_session.dataframe.DataFrame.write.mode().partitionBy().option().csv.assert_called_with("gs://test")
+        
+    @mock.patch.object(pyspark.sql, 'SparkSession')       
+    @mock.patch.object(logging, 'Logger')     
+    def test_write_data_parquet(self, Logger, mock_spark_session):
+        """Tests SnowflakeToGCSTemplate to write data"""    
+        
+        snowflake_to_gcs_template = SnowflakeToGCSTemplate()
+        
+        mock_parsed_args = snowflake_to_gcs_template.parse_args(
+            ["--snowflake.to.gcs.sf.url=url", 
+             "--snowflake.to.gcs.sf.user=user", 
+             "--snowflake.to.gcs.sf.password=password", 
+             "--snowflake.to.gcs.sf.database=database",
+             "--snowflake.to.gcs.sf.warehouse=dwh",
+             "--snowflake.to.gcs.sf.query=select 1", 
+             "--snowflake.to.gcs.output.location=gs://test", 
+             "--snowflake.to.gcs.output.format=parquet", 
+             "--snowflake.to.gcs.sf.autopushdown=yes",
+             "--snowflake.to.gcs.output.mode=append"
+             ]
+        )
+        
+        mock_spark_session.read.format().options().option().load.return_value = mock_spark_session.dataframe.DataFrame
+        snowflake_to_gcs_template.write_data(Logger, mock_parsed_args, mock_spark_session.dataframe.DataFrame)
+        mock_spark_session.dataframe.DataFrame.write.mode.assert_called_with(constants.OUTPUT_MODE_APPEND)
+        mock_spark_session.dataframe.DataFrame.write.mode().parquet.assert_called_with("gs://test")
+        
+    @mock.patch.object(pyspark.sql, 'SparkSession')       
+    @mock.patch.object(logging, 'Logger')     
+    def test_write_data_json(self, Logger, mock_spark_session):
+        """Tests SnowflakeToGCSTemplate to write data"""   
+        
+        snowflake_to_gcs_template = SnowflakeToGCSTemplate()
+        
+        mock_parsed_args = snowflake_to_gcs_template.parse_args(
+            ["--snowflake.to.gcs.sf.url=url", 
+             "--snowflake.to.gcs.sf.user=user", 
+             "--snowflake.to.gcs.sf.password=password", 
+             "--snowflake.to.gcs.sf.database=database",
+             "--snowflake.to.gcs.sf.warehouse=dwh",
+             "--snowflake.to.gcs.sf.query=select 1", 
+             "--snowflake.to.gcs.output.location=gs://test", 
+             "--snowflake.to.gcs.output.format=json", 
+             "--snowflake.to.gcs.sf.autopushdown=yes",
+             "--snowflake.to.gcs.output.mode=append"
+             ]
+        )
+        
+        mock_spark_session.read.format().options().option().load.return_value = mock_spark_session.dataframe.DataFrame
+        snowflake_to_gcs_template.write_data(Logger, mock_parsed_args, mock_spark_session.dataframe.DataFrame)
+        mock_spark_session.dataframe.DataFrame.write.mode.assert_called_with(constants.OUTPUT_MODE_APPEND)
```

Comment:
Add an "assert_called_with" the json format

Reply:
Line no.301 is performing an assert_called_with on json(). Is it different from what you're expecting?

Reply:
Sure, you are right, I did not see that. Resolved.